cypress-io/cypress
# breaking: upgrade cy.readFile() to be a query command

### Additional details
This PR updates `cy.readFile()` to be a query command. It is a breaking change because users will no longer be able to overwrite `readFile` with `Cypress.Commands.overwrite()`. No other impact on user's tests is anticipated.

`cy.readFile()` already matched the behavior of other queries, by reading the file from disk until upcoming assertions pass. This update therefore is not a behavior change - it simply brings the readFile code execution path in line with other queries which do the same thing.

```
// Rereads file from disk until assertion passes
cy.readFile('foo.json').should('eql', { user: 'bar', id: 123 })

// Requeries element from DOM until assertion passes
cy.get('#div').should('have.class', 'active')

// Does not reread file from disk
cy.fixture('foo.json').should('eql', { user: 'bar', id: 123 })
```

Despite there being no user-facing changes, this update is a prerequisite for https://github.com/cypress-io/cypress/pull/25296. In that pull request, assertions will no longer "belong" to previous commands - they execute entirely independently. That means that the *only* way for a command to update the subject in response to failed assertions is to be a query - .readFile() can no longer be a special case of a "query-like command that isn't a query."

When reviewing / aiming to understand the changes, suggest starting with this comment: https://github.com/cypress-io/cypress/pull/25595#discussion_r1088141640. 

### Steps to test
No behavior has changed, and tests required no updates (see comments on each individual change for why it is code-clarity related and not functional).

### How has the user experience changed?
Previously, if readFile timed out, users would get the message

```
`cy.readFile(foo.txt)` timed out retrying after 4000ms.
```

They now instead get the message

```
Timed out retrying after 4000ms: `cy.readFile(foo.txt)` timed out.
```

### PR Tasks

- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/5017
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
